Health/blood bar design #4

ykoval201 opened the pull request:
Health bar design and script. made into a prefab

ykoval201 committed:
Added Health bar

Added Health bar attached to a circle with player script.
Health bar functions.
SPACE to deal 20 damage
H to heal 20 damage

ykoval201 committed:
HB Sprite update

ykoval201 committed:
small position adj

ykoval201 committed:
Removed code

Removed code not related to the issue.

ykoval201 committed:
Merge remote-tracking branch 'origin/premain' into 4_Health/bloodBarDesign

ykoval201 committed:
Made a prefab of a health bar

Added Health bar prefab into prefabs folder

adenprince commented on Assets/Player.cs:
```diff
@@ -0,0 +1,49 @@
+using UnityEngine;
+using UnityEngine.UI;
+
+public class Player : MonoBehaviour
```
- [x] The functionality here needs to go in the `Assets/Scripts/Player/PlayerController.cs` script.

adenprince replied:
- [x] Remove this script.

adenprince commented on Assets/Player.cs:
```diff
@@ -0,0 +1,49 @@
+    // Update is called once per frame
+    void Update()
+    {
+        //Restore health to max when H is pressed
```
It's not that big of a deal, but this comment seems inaccurate. I think having a comment here (and above `void Update()`) is unnecessary.

adenprince commented on Assets/Player.cs:
```diff
@@ -0,0 +1,49 @@
+using System.Collections;
```
It would be good to remove the unused using directives in `Player.cs` and `HealthBar.cs`.

adenprince commented on Assets/Player.cs:
```diff
@@ -0,0 +1,49 @@
+
+    public void takeDamage(int damage)
+    {
+        currentHealth -= damage;
```
- [x] There should be a check in `takeDamage` that prevents `currentHealth` from going below 0 and a check in `heal` that prevents `currentHealth` from going above `maxHealth`.

adenprince commented on ProjectSettings/VersionControlSettings.asset:
```diff
@@ -3,6 +3,6 @@
 VersionControlSettings:
   m_ObjectHideFlags: 0
-  m_Mode: Visible Meta Files
+  m_Mode: Perforce
```
I'm not sure why this got changed. This change should be reverted.
- [x] Revert

adenprince commented on Assets/Player.cs:
```diff
@@ -0,0 +1,49 @@
+
+public class Player : MonoBehaviour
+{
+    public int maxHealth = 100;
```
It's better to use `[SerializeField] private` instead of `public` for variables that appear in the Editor. Then, instance variables are not automatically available outside of their class.

adenprince commented on Assets/Scenes/SampleScene.unity:
```diff
@@ -200,67 +200,399 @@ Transform:
   m_RootOrder: 0
   m_LocalEulerAnglesHint: {x: 0, y: 0, z: 0}
---- !u!1001 &1227330104
+--- !u!1 &1433527202
```
- [x] You should add the Player prefab back to SampleScene and remove the Player GameObject you added to that scene after moving the health code to `PlayerController.cs`. Then, player movement and health will both be in SampleScene.

adenprince commented on UserSettings/Layouts/CurrentMaximizeLayout.dwlt:
```diff
@@ -0,0 +1,927 @@
+%YAML 1.1
```
User setting files are not tracked anymore with a `.gitignore` update that occurred after you added this file. You can delete this file and then push that deletion to remove it. (It will get generated again by the Unity Editor.) (Here is how you can remove the file from the repository without deleting it locally: https://stackoverflow.com/a/1143800.)
- [x] Remove this file

ykoval201 replied:
I did the git command but I'm not sure if that worked.

adenprince replied:
It looks correct to me. I don't see this file on your branch anymore.

adenprince commented on Assets/Sprites/Health bar 4.png.meta:
```diff
@@ -0,0 +1,122 @@
+fileFormatVersion: 2
```
`Health bar 4.png` could be renamed to `Health Bar Border.png` for the file name to be more clear, but it doesn't really matter much.

adenprince commented on Assets/Player.cs:
```diff
@@ -0,0 +1,49 @@
+        currentHealth += heal;
+        healthBar.SetHealth(currentHealth);
+    }
+
```
There are some extra empty lines in your script changes. These should be removed, but it's not that big of a deal.

adenprince commented on Assets/HealthBar.cs:
```diff
@@ -0,0 +1,21 @@
+using System.Collections;
```
- [x] Move this script to the `Assets/Scripts` folder. (Maybe to `Assets/Scripts/UI`.)

adenprince commented on Assets/Prefabs/Main Player health bar.prefab:
```diff
@@ -0,0 +1,258 @@
+%YAML 1.1
```
The health bar fill doesn't scale correctly horizontally with the outline. You can see this by making the Game window smaller horizontally. I think this is caused by the fill having a narrower width than the outline, but it's just a minor cosmetic problem. One solution would be to create and use a fill sprite with the same size as the outline sprite.

adenprince reviewed (changes requested):
I made tasks (empty checkboxes) for changes that are needed before merging to premain, and I made some other suggestions in comments without a task. Your branch name is a little different from the format specified in `CONTRIBUTING.md`, but I'm fine with it because it contains the issue number and an issue description.

ykoval201 committed:
Pull request issues

-Moved the functionality of Player.cs to PlayerController.cs
- Removed unused using directives in HealthBar.cs
-removed Player.cs entirely
-added checks to prevent health from going below 0 and above 100
- removed CurrentMaximizeLayout.dwlt using git command
- small changes in code.

adenprince commented on Assets/Scripts/Player/PlayerController.cs:
```diff
@@ -3,13 +3,19 @@
 
     private Rigidbody2D rb2D;
     private Vector2 targetPosition;
+    public HealthBar healthBar;
```
- [x] Use `[SerializeField] private` for `healthBar`.

adenprince commented on Assets/Scripts/Player/PlayerController.cs:
```diff
@@ -31,4 +47,20 @@ public class PlayerController : MonoBehaviour {
+    }
+    public void heal(int heal)
+    {
+        if (currentHealth < maxHealth)
```
- [x] If the player is at 90 health and gets healed for 20 health, `currentHealth` will go above `maxHealth`. If the player is at 10 health and gets damaged for 20 health, `currentHealth` will go below 0. These errors can be fixed by doing the increment or decrement first and then setting `currentHealth` to the limit if it goes past the limit.

adenprince commented on Assets/Scripts/UI/HealthBar.cs:
```diff
@@ -0,0 +1,19 @@
+public class HealthBar : MonoBehaviour
+{
+
+    public Slider slider;
```
- [x] Change `public` to `[SerializeField] private`. (Also, line 6 is an extra newline.)

adenprince commented on Assets/Scripts/Player/PlayerController.cs:
```diff
@@ -31,4 +47,20 @@ public class PlayerController : MonoBehaviour {
         // TODO: Maybe set targetPosition to currentPosition when the player is stopped. Then, the player
         // won't continually try to reach an unreachable targetPosition when blocked by a collider.
     }
+    public void takeDamage(int damage)
```
It would be more consistent for these method names to be capitalized and for the method definitions to have a newline between them, but it doesn't matter much right now. Hopefully, we can write down style guidelines in the future so that they are not so unclear.

ykoval201 replied:
Done

adenprince commented on Assets/Scenes/SampleScene.unity:
```diff
@@ -176,7 +176,7 @@ Camera:
   field of view: 60
   orthographic: 1
-  orthographic size: 5
+  orthographic size: 1
```
In my opinion, the previous camera size (5) in `SampleScene` is too zoomed out, and the changed camera size (1) in that scene is too zoomed in. It's up to you whether to change the camera size or not because the camera size is unrelated to this pull request.

ykoval201 replied:
I set it to 3.

adenprince reviewed (changes requested):
Good job on making the requested changes. I made tasks for a few small problems that need to be fixed. Once those tasks are completed, this pull request should be ready to merge. 👍

ykoval201 committed:
Additional pull request issues

- Remove Player.cs
- serialized variables in PlayerController and HealthBar
- modified check conditions for damge and heal functions.
- Capitalized names of functions
-Changed camera size to 3

adenprince commented on Assets/Scripts/Player/PlayerController.cs:
```diff
@@ -3,13 +3,20 @@
     private Rigidbody2D rb2D;
     private Vector2 targetPosition;
+    [SerializeField]
+    public HealthBar healthBar;
```
- [ ] Make this variable private.

adenprince replied:
- [x] Make this variable private.

adenprince commented on Assets/Scenes/SampleScene.unity:
```diff
@@ -176,7 +176,7 @@ Camera:
   field of view: 60
   orthographic: 1
-  orthographic size: 5
+  orthographic size: 1
```
It seems like the camera size change didn't get pushed.

adenprince reviewed (changes requested):
I added one small cleanup task. Once that's complete, this pull request should be good to merge.

ykoval201 committed:
Small issues

made healthbar private
hopefully changed camera size
